### PR TITLE
Add end-to-end tests

### PR DESCRIPTION
**Pull request by silvin-lubecki**
* Add --insecure flag to share bundle using plain HTTP
* Fix error messages
* Add e2e test to push/pull/fixup a CNAB to a registry

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,44 @@
+ARG ALPINE_VERSION=3.8
+ARG GO_VERSION=1.11.0
```
Comment (chris-crone) [inline Dockerfile]: Bump to most recent 1.11

Comment (chris-crone) [reply Dockerfile]: Don't see either of these ARGs used

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+# build image
+FROM golang:alpine as build
```
Comment (chris-crone) [inline Dockerfile]: ARGs should be used here I think

File: e2e/run.sh [hunk 3]
```diff
@@ -0,0 +1,6 @@
+#!/usr/bin/env sh
+set -eu -o pipefail
+
+# Run the e2e tests
+cd ./e2e
+./e2e.test
```
Comment (chris-crone) [inline e2e/run.sh]: nit: \n

File: Dockerfile (hunk 2 above)
Comment (silvin-lubecki) [reply Dockerfile]: Obviously 😅

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+RUN curl -Ls https://github.com/docker/app/releases/download/$DOCKER_APP_VERSION/docker-app-linux.tar.gz | tar -xz
+RUN git clone https://github.com/docker/app
+
+WORKDIR /go/src/github.com/docker/cnab-to-oci
+COPY . .
+RUN make bin/cnab-to-oci &&\
+    make build-e2e-test
+
+# e2e image
+FROM alpine as e2e
```
Comment (chris-crone) [inline Dockerfile]: nit: Should use the ARG here too

**Pull request merged**